example/project/cluster/integration/bootstrap_test.lua

```diff
@@ -0,0 +1,217 @@
+        server_command = helper.server_command,
+        use_vshard = true,
+        replicasets = replicasets
+    })
+
+    group.cluster:start()
+    group.cluster:upload_config({
+        metrics = {
+            export = {
+                {
+                    path = '/metrics',
+                    format = 'json'
+                },
+                {
+                    path = '/metrics/prometheus',
+                    format = 'prometheus'
+                }
+            }
+        }
+    })
+end)
+
+test.after_suite(function()
+    group.cluster:stop()
+end)
+
+local function http_request(alias, method, endpoint, count)
+    if count <= 0 then
+        return
+    end
+
+    for _ = 1, count do
+        group.cluster:server(alias):http_request(method, endpoint, { raise = false })
+    end
+end
+
+local last_key = {
+    ['tnt-router'] = 1,
+    ['tnt-storage-1-master'] = 1,
+    ['tnt-storage-2-master'] = 1,
+}
+
+local charset = {} -- [0-9a-zA-Z]
+for c = 48, 57  do table.insert(charset, string.char(c)) end
+for c = 65, 90  do table.insert(charset, string.char(c)) end
+for c = 97, 122 do table.insert(charset, string.char(c)) end
+
+local function random_string(length)
+    if not length or length <= 0 then return '' end
+    math.randomseed(os.clock()^5)
+    return random_string(length - 1) .. charset[math.random(1, #charset)]
+end
+
+local function space_operations(alias, operation, count)
+    if count <= 0 then
+        return
+    end
+
+    local space = group.cluster:server(alias).net_box.space.MY_SPACE
+
+    if operation == 'select' then
+        for _ = 1, count do
+            space[operation](space, {}, { limit = 1 })
+        end
+
+    elseif operation == 'insert' then
+        for _ = 1, count do
+            space[operation](space, { last_key[alias], random_string(5) })
+            last_key[alias] = last_key[alias] + 1
+        end
+
+    elseif operation == 'update' then
+        for _ = 1, count do
+            local key = space:select({}, { limit = 1 })[1][1]
+            space[operation](space, key, {{ '=', 2, random_string(5) }})
+        end
+
+    elseif operation == 'upsert' then
+        for _ = 1, count do
+            local tuple = space:select({}, { limit = 1 })[1]
+            space[operation](space, tuple, {{ '=', 2, random_string(5) }})
+        end
+
+    elseif operation == 'replace' then
+        for _ = 1, count do
+            local key = space:select({}, { limit = 1 })[1][1]
+            space[operation](space, { key, random_string(5) })
+        end
+
+    elseif operation == 'delete' then
+        for _ = 1, count do
+            local key = space:select({}, { limit = 1 })[1][1]
+            space[operation](space, { key })
+        end
+    end
+end
+
+group.test_cluster = function()
+    test.helpers.retrying({ timeout = math.huge },
+        function()
+            -- Generate some HTTP traffic
+            http_request('tnt-router', 'get', '/hello', math.random(5, 10))
```

Comment:
string literals to identify objects or operation type often leads to typos. try to offload linters job to linter. firsts argument could be replaced with actual instance object. second argument must be replaced with constant (e.g. enum).